### PR TITLE
Organize defaultsettings.cpp

### PR DESCRIPTION
**Pull request by ShadowNinja**
Split out of #2604.

### CODE REVIEW

**Comment by nerzhul** [comment]
Seems interesting. @minetest/team-minetest please review

**Comment by est31** [comment]
I don't really care about the change itself, but yeah its better if its split out of the PR.

**Comment by paramat** [comment]
Mapgen settings are wrong, investigating.

File: src/defaultsettings.cpp [hunk 1]
```diff
@@ -321,27 +323,27 @@ void set_default_settings(Settings *settings)
...
 	settings->setDefault("liquid_queue_purge_time", "0");
 	settings->setDefault("liquid_update", "1.0");
 
-	//mapgen stuff
+	// Mapgen
 	settings->setDefault("mg_name", "v7");
 	settings->setDefault("water_level", "1");
 	settings->setDefault("chunksize", "5");
 	settings->setDefault("mg_flags", "dungeons");
+	settings->setDefault("mgv6_spflags", "jungles, snowbiomes");
```
Comment (paramat) [inline src/defaultsettings.cpp]: Where did this come from? It's not present in current settings and not present in MTGame .conf.

File: src/defaultsettings.cpp [hunk 2]
```diff
@@ -321,27 +323,27 @@ void set_default_settings(Settings *settings)
...
+	settings->setDefault("fixed_map_seed", "");
+	settings->setDefault("enable_floating_dungeons", "true");
```
Comment (paramat) [inline src/defaultsettings.cpp]: AFAIK this setting doesn't exist anymore, checking.

Comment (paramat) [inline src/defaultsettings.cpp]: Setting was removed when i fixed floating dungeon slices https://github.com/minetest/minetest/commit/17b7b7c85fc4faca2180a0104ed457c3a8734627

**Comment by ShadowNinja** [comment]
@paramat: Removed those two settings, they were artifacts of a rebase from over a year ago.

File: src/defaultsettings.cpp (hunk 1 above)
Comment (paramat) [inline src/defaultsettings.cpp]: Removed here https://github.com/minetest/minetest/commit/ebe638c32db2fb29886a5ae82c7726afad4c0010

**Comment by paramat** [comment]
Since so much has changed since April 2015 is it not better to completely redo this commit? There are 2 mistakes in mapgen stuff so there may be others.

**Comment by paramat** [comment]
Ok mapgen stuff looks ok now.

File: src/defaultsettings.cpp [hunk 3]
```diff
@@ -56,12 +59,11 @@ void set_default_settings(Settings *settings)
 	settings->setDefault("keymap_toggle_hud", "KEY_F1");
 	settings->setDefault("keymap_toggle_chat", "KEY_F2");
 	settings->setDefault("keymap_toggle_force_fog_off", "KEY_F3");
-	settings->setDefault("keymap_toggle_update_camera",
+	settings->setDefault("keymap_toggle_update_camera", ""
```
Comment (Zeno-) [inline src/defaultsettings.cpp]: Please don't do this. I think it's better to have more typing and have both entire lines in the body of #if and #else

**Comment by ShadowNinja** [comment]
@paramat: I've now looked over all of the changed lines both manually and semi-automated (load patch, select all setDefault lines, remove +/-, sort, and make sure there are exactly two duplicates for every line).

File: src/defaultsettings.cpp (hunk 3 above)
Comment (ShadowNinja) [inline src/defaultsettings.cpp]: Done.

**Comment by paramat** [comment]
map_generation_limit should be moved to mapgen stuff.

**Comment by paramat** [comment]
And max_block_generate_distance too.

**Comment by paramat** [comment]
^ Both of those are under mapgen in .conf.

**Comment by paramat** [comment]
I'm not insisting on starting this again, just suggesting it if you find that helpful to reduce errors.

**Comment by paramat** [comment]
Settingtypes.txt / minetest.conf might be helpful in suggesting organisation, however without splitting stuff into 'advanced' categories.

**Comment by ShadowNinja** [comment]
I've done some additional organization.  In the future we may want to use the settingtypes.txt order, but that's more than I want to do right now. :-)

**Comment by paramat** [comment]
Mapgen stuff is fine now, the rest looks good but i haven't thoroughly checked the changes.

**Comment by paramat** [comment]
:+1:

**Pull request merged**